pkg/controller/datavolume-controller.go

```diff
@@ -1483,6 +1605,59 @@ func newSnapshot(dataVolume *cdiv1.DataVolume, snapshotName, snapshotClassName s
 	return snapshot
 }
 
+// NewVolumeClonePVC creates a PVC object to be used during CSI volume cloning. csiClonePvcType defines whether the PVC object will be a source cloner PVC or the target PVC
+func newVolumeClonePVC(dv *cdiv1.DataVolume, sourcePvc *corev1.PersistentVolumeClaim, targetPvcSpec *corev1.PersistentVolumeClaimSpec, pvcName string) *corev1.PersistentVolumeClaim {
```

Comment:
Can we just call newPersistentVolumeClaim, and override the pvc.Spec.DataSource before returning the result. That way all the logic that sets annotations and ownerRefs and everything else is contained in that function.

Reply:
I can try, we will see if that is better

Reply:
Done